stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/Forward.java

```diff
@@ -63,14 +64,20 @@ protected HttpCarbonMessage createOutboundRequestMsg(Context context) {
 
         if (HttpUtil.isEntityDataSourceAvailable(requestStruct)) {
             HttpUtil.enrichOutboundMessage(outboundRequestMsg, requestStruct);
-            prepareOutboundRequest(context, path, outboundRequestMsg);
+            prepareOutboundRequest(context, path, outboundRequestMsg, isNonEntityBodyReq(outboundRequestMsg));
             outboundRequestMsg.setProperty(HttpConstants.HTTP_METHOD,
                     BLangConnectorSPIUtil.toStruct(requestStruct).getStringField(HttpConstants.HTTP_REQUEST_METHOD));
         } else {
-            prepareOutboundRequest(context, path, outboundRequestMsg);
+            prepareOutboundRequest(context, path, outboundRequestMsg, isNonEntityBodyReq(outboundRequestMsg));
             String httpVerb = (String) outboundRequestMsg.getProperty(HttpConstants.HTTP_METHOD);
             outboundRequestMsg.setProperty(HttpConstants.HTTP_METHOD, httpVerb.trim().toUpperCase(Locale.getDefault()));
         }
         return outboundRequestMsg;
     }
+
+    private Boolean isNonEntityBodyReq(HttpCarbonMessage outboundRequestMsg) {
```

Comment:
method name sounds a bit weird same as in NON_ENTITY_BODY_REQUEST constant

Reply:
The same elaboration applies here as well.